synapse/storage/databases/main/registration.py

```diff
@@ -110,6 +110,7 @@ async def get_user_by_id(self, user_id: str) -> Optional[Dict[str, Any]]:
                 "creation_ts",
                 "user_type",
                 "deactivated",
+                "shadow_banned",
```

Comment:
The cache on `get_user_by_id` needs to be invalidated in `set_shadow_banned`, it would be something like:

```
self._invalidate_cache_and_stream(
    txn, self.get_user_by_id, (user.to_string(),)
)
```